--- pm1fs2.c.old	2010-02-09 09:34:10.000000000 +0100
+++ pm1fs2.c	2010-02-09 09:34:32.000000000 +0100
@@ -1334,7 +1334,8 @@
     mpmod_copy (modulus_local, modulus);
 #pragma omp for
 #else
-    mpmod_t modulus_local = modulus;
+    mpmod_t modulus_local;
+    mpmod_copy (modulus_local, modulus);
 #endif
     for (i = 0; (unsigned long) i <= 2 * deg - 2; i++)
       mpres_mul_z_to_z (H[i], Vt, H[i], modulus_local);
